examples/gps_bot/gps_bot.cc

```diff
@@ -0,0 +1,90 @@
+// BoB robotics includes
+#include "common/logging.h"
+#include "common/timer.h"
+#include "common/gps.h"
+#include "hid/joystick.h"
+#include "robots/norbot.h"
+#include "third_party/units.h"
+
+// Standard C++ includes
+#include <iostream>
+#include <thread>
+
+// Standard C includes
+#include <cmath>
+
+using namespace BoBRobotics;
+using namespace units::angle;
+using namespace units::length;
+
+void gpsThreadFunc(const char *device, std::mutex &mutex, degree_t &latOut, degree_t &lonOut, meter_t &altitudeOut)
+{
+    Gps gps("/dev/ttyACM0");
+
+    std::cout << "GPS created" << std::endl;
+    while(true)
```

Comment:
Won't this thread run forever? I would use a ``std::atomic_bool`` as a stop flag, then you can set it to false on your main thread, then join this thread while you wait for it to finish.

Reply:
Yes...I am to blame for the threading code in this - it was written at Capo Caccia, late at night :smile: